app/models/miq_queue.rb

```diff
@@ -153,6 +153,27 @@ def self.put(options)
     msg
   end
 
+  # Execute a job on all servers.
+  #
+  # Will clear out :zone and :role from arguments (allowing to target all
+  # active servers), and even raise an error if they
+  #
+  def self.broadcast(options)
+    server_targets = MiqServer.active_miq_servers.select(:id, :guid)
+
+    # Currently not filterable by these keys (:zone, :role)
+    #
+    # If this feature is ever needed, remove these errors and filter the
+    # servers above based ones that match the filters, as well as passing them
+    # in the `.put` params below.
```

Comment:
I do, as it is important to not just pass the raw params to `.put`, otherwise you will be creating jobs that will never get executed.

That said,  I will be re-wording it to emphasize that point better.